File: helix/src/main/java/glitch/helix/object/json/list/Videos.java
package glitch.helix.object.json.list;

import com.google.common.collect.ImmutableList;
import com.google.gson.annotations.SerializedName;
import glitch.api.objects.json.interfaces.CursorList;
import glitch.helix.object.json.Video;
import lombok.Data;

@Data
public class Videos implements CursorList<Video> {
    private final ImmutableList<Video> data;

    @SerializedName("pagination.cursor")
    private final String cursor;
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/collections/Videos.kt
package glitch.kraken.`object`.json.collections

import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.OrdinalList
import glitch.kraken.`object`.json.Video

data class Videos(
        override val data: List<Video>,
        @SerializedName("total")
        override val size: Int = data.size
) : OrdinalList<Video>